logic/out_transfer.js

```diff
@@ -89,9 +89,7 @@ OutTransfer.prototype.verify = function(transaction, sender, cb) {
 		return setImmediate(cb, 'Invalid recipient');
 	}
 
-	const amount = new Bignum(transaction.amount);
-
-	if (!amount.equals(0)) {
+	if (!transaction.amount) {
```

Comment:
@nazarhussain I think this will not work, `if (!transaction.amount) {` will always result in `false` as `new BigNum(0)` will be `true`. The condition should be `if (transaction.amount.equals(0)`. Also can you ensure once that `transaction.amount` is bignumber?

Reply:
reference for the implementation here https://runkit.com/5baa0981ae64c20012b8228b/5bb234611992d60012821384

Reply:
I believe `transaction.amount` at this context is not a big number. So just checking if `transaction` have an `amount` attribute is good enough. Its exactly the old logic, I just reverted it back.

Reply:
I think it's Bignumber, in logic transaction object normalize the amount will be converted to Bignumber, so before it goes to any transaction verify step it will be converted to Bignumber, here is the logic
https://github.com/LiskHQ/lisk/blob/ca8a69fdaa4d69ec729721a3b49b04172a337dff/logic/transaction.js#L1126

Reply:
If that's the case then why we were casting it to `bignum` in the `verify` method.